p2p/net/swarm/swarm.go

```diff
@@ -209,8 +211,12 @@ func NewSwarm(local peer.ID, peers peerstore.Peerstore, eventBus event.Bus, opts
 	}
 
 	s.dsync = newDialSync(s.dialWorkerLoop)
+
 	s.limiter = newDialLimiter(s.dialAddr)
 	s.backf.init(s.ctx)
+
+	s.bhd = newBlackHoleDetector(true, true, s.metricsTracer)
```

Comment:
We need a config option for this. Ideally, the user could configure the thresholds. Doesn't need to be part of this PR, but we'd need this for the next release.

Reply:
you mean with libp2p.New?

Reply:
Same as for the dial ranker.

Reply:
will raise a separate PR